Lib/collections/__init__.py

```diff
@@ -412,26 +368,62 @@ def namedtuple(typename, field_names, *, verbose=False, rename=False, module=Non
             raise ValueError('Encountered duplicate field name: %r' % name)
         seen.add(name)
 
-    # Fill-in the class template
-    class_definition = _class_template.format(
-        typename = typename,
-        field_names = tuple(field_names),
-        num_fields = len(field_names),
-        arg_list = repr(tuple(field_names)).replace("'", "")[1:-1],
-        repr_fmt = ', '.join(_repr_template.format(name=name)
-                             for name in field_names),
-        field_defs = '\n'.join(_field_template.format(index=index, name=name)
-                               for index, name in enumerate(field_names))
+    arg_list = repr(tuple(field_names)).replace("'", "")[1:-1]
+    num_fields = len(field_names)
+    repr_fmt = '(' + ', '.join(_repr_template.format(name=name) for name in field_names) + ')'
+
+    namespace = {'_tuple': tuple}
+    exec(_new_template.format(typename=typename, arg_list=arg_list), namespace)
+    __new__ = namespace['__new__']
+
+    @classmethod
+    def _make(cls, iterable, new=tuple.__new__, len=len):
+        result = new(cls, iterable)
+        if len(result) != num_fields:
```

Comment:
This will keep the closure alive, maybe make ``num_fileds`` a (private) attribute of the class and define this function outside of ``namedtuple``? (TBH however, I have no idea how efficient this will be, but intuitively this should be more efficient than with nested function at least in terms of memory).